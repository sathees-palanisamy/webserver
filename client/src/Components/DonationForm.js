import React from "react";
import { useSelector, useDispatch } from "react-redux";
import * as actions from "../Store/actions";
import Nav from "./Nav";

function DonationForm(props) {
  const [userId, setUserId] = React.useState("");
  const [donationAmt, setDonationAmt] = React.useState(0);
  const [tipAmount, setTipAmount] = React.useState(0);
  const [formError,setFormError] = React.useState("");

  const errorDetail = useSelector((state) => state.donation.insertStatus);
  const dispatch = useDispatch();

  // console.log("errorDetail:" + errorDetail);

  const saveDonation = (event) => {
    event.preventDefault();
    console.log(userId);
    console.log(donationAmt);
    console.log(tipAmount);

    if (userId !== "") {
      setFormError("");
      setUserId("");
      setDonationAmt(0);
      setTipAmount(0);
      dispatch(actions.insertInPorgress());
      dispatch(actions.postDonationItem(userId, donationAmt, tipAmount));
    } else {
      setFormError("Invalid Input")
    }
  };

  const handleUserId = (event) => {
    setUserId(event.target.value);
  };

  const handleDonationAmt = (event) => {
    setDonationAmt(event.target.value);
  };

  const handleTipAmt = (event) => {
    setTipAmount(event.target.value);
  };

  return (
    <div className='container' data-test='form-container'>
      <h1 data-test='form-header'>Form Details</h1>
   
      <form onSubmit={saveDonation} className="Form" data-test='form-detail'>
        <div className="Form--field" data-test='userid-field-details'>
          <label data-test='userid-field-label'>User ID: </label>
          <input
            type="text"
            onChange={handleUserId}
            id="userId"
            value={userId}
            name="userId"
            data-test='userid-field-input'
            className="input-userid"
          />
        </div>

        <div className="Form--field" data-test="amount-field-details">
          <label data-test="amount-field-label">Donation Amount: </label>
          <input
            type="number"
            onChange={handleDonationAmt}
            id="amount"
            value={donationAmt}
            data-test="amount-field-input"
          />
        </div>
        <div className="Form--field" data-test="tip-field-details">
          <label data-test="tip-field-label">Tip Amount: </label>
          <input
            type="number"
            onChange={handleTipAmt}
            id="amount"
            value={tipAmount}
            data-test="tip-field-input"
          />
        </div>
        <button className="Form__button" type="submit" data-test="form-button">
          Add Donation
        </button>
      </form>
      <div>{errorDetail}</div>
      <div data-test="form-error">{formError}</div>
    </div>
  );
}

export default DonationForm;
