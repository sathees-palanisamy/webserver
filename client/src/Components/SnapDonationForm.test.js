import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import { storeFactory } from '../test/testutil';
import DonationForm from './DonationForm';

const store = storeFactory({});

it('renders form details', () => {
  const tree = renderer.create(<Provider store={store}><DonationForm  /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});