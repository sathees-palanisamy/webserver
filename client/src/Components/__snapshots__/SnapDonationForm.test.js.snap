// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form details 1`] = `
<div
  className="container"
  data-test="form-container"
>
  <h1
    data-test="form-header"
  >
    Form Details
  </h1>
  <form
    className="Form"
    data-test="form-detail"
    onSubmit={[Function]}
  >
    <div
      className="Form--field"
      data-test="userid-field-details"
    >
      <label
        data-test="userid-field-label"
      >
        User ID: 
      </label>
      <input
        className="input-userid"
        data-test="userid-field-input"
        id="userId"
        name="userId"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="Form--field"
      data-test="amount-field-details"
    >
      <label
        data-test="amount-field-label"
      >
        Donation Amount: 
      </label>
      <input
        data-test="amount-field-input"
        id="amount"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </div>
    <div
      className="Form--field"
      data-test="tip-field-details"
    >
      <label
        data-test="tip-field-label"
      >
        Tip Amount: 
      </label>
      <input
        data-test="tip-field-input"
        id="amount"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </div>
    <button
      className="Form__button"
      data-test="form-button"
      type="submit"
    >
      Add Donation
    </button>
  </form>
  <div>
    
  </div>
  <div
    data-test="form-error"
  >
    
  </div>
</div>
`;
